README.md

```diff
@@ -231,10 +231,59 @@ iex> PersonTranscoder.encode!(%{name: "Jane Doe",
 
 This can be used to, for example, build Ecto changesets via a `changeset/2` functions and to render HAL responses to HTTP requests.
 
+#### Patching
+
+ExHal.Transcoder also supports modifying objects with [JSON patch](https://tools.ietf.org/html/rfc6902).
+
+```elixir
+defmodule PetTranscoder do
+  use ExHal.Transcoder
+
+  defproperty "name"
+  defproperty "animalType",   param: species, protected: true
+  deflink     "favoriteToy",  param: :favorite_toy
+  deflinks    "friends"
+end
+```
+
+Create an object from the transcoder, then modify it
+
+```elixir
+iex> spot = PetTranscoder.decode!(doc)
+%{name: "Spot",
+  species: "dog",
+  favorite_toy: "http://a.co/56guxwO"
+  friends: ["https://petbook.com/u/fifi", "https://petbook.com/u/fido"]}
```

Comment:
oic... then patch document is wrong, i think.

Given a JSON representation:

```json
{ "name": "Jane Doe",
   "mailingAddress": "123 Main St",
   "_links": {
      "app:department": { "href": "http://example.com/dept/42" },
      "tags": [
        {"href": "foo:1"}
      ]
  }
}
```
If you want to add a tag would be

```json
[ { "op": "add", "path": "/_links/tags/-", "value": { "href": "urn:x" } } ] 
```